nixos-generate.nix

```diff
@@ -17,12 +21,21 @@ let
       };
     };
   };
+
+  # Will only get evaluated when used, so no worries
+  flake = builtins.getFlake flakeUri;
+  flakeSystem = flake.outputs.packages."${system}".nixosConfigurations."${flakeAttr}" or flake.outputs.nixosConfigurations."${flakeAttr}";
 in
-import "${toString nixpkgs}/nixos/lib/eval-config.nix" {
-  inherit system;
-  modules = [
-    module
-    format-config
-    configuration
-  ];
-}
+  if flakeUri != null then
+    flakeSystem.override (attrs: {
+      modules = attrs.modules ++ [ module formatConfig ];
+    })
+  else
+    import "${toString nixpkgs}/nixos/lib/eval-config.nix" {
```

Comment:
I think this is equivalent to `nixpkgs.path`? I'm not sure when it was introduced though.

Reply:
Yeah it's definitely not new. I thought it looked a wee bit strange, but I ignored it